app/workers/chase_references.rb

```diff
@@ -1,75 +1,87 @@
+    #
+    if chaser[:old_type]
+      rejected_chased_ids = rejected_chased_ids.or(ChaserSent.send(chaser[:old_type]).select(:chased_id))
     end
-  end
 
-  def send_14_day_chaser!
-    references = ApplicationReference
-      .joins(:application_form)
-      .feedback_requested
-      .where(application_forms: { recruitment_cycle_year: ApplicationForm.select('candidate_id').maximum(:recruitment_cycle_year) })
-      .where(['requested_at < ?', TimeLimitConfig.replace_referee_by.days.before(Time.zone.now)])
-      .where.not(id: ChaserSent.reference_replacement.select(:chased_id))
+    references = ApplicationReference.referees_to_chase(chase_referee_by: chase_referee_by, rejected_chased_ids: rejected_chased_ids)
 
     references.each do |reference|
-      SendNewRefereeRequestEmail.call(
-        reference: reference,
-        reason: :not_responded,
-      )
-    end
-  end
-
-  def send_21_day_chaser!
-    references = ApplicationReference
-                   .joins(:application_form)
-                   .feedback_requested
-                   .where(application_forms: { recruitment_cycle_year: ApplicationForm.select('candidate_id').maximum(:recruitment_cycle_year) })
-                   .where(['requested_at < ?', TimeLimitConfig.second_chase_referee_by.days.before(Time.zone.now)])
-                   .where.not(id: ChaserSent.reminder_reference_nudge.select(:chased_id))
+      deliver_email(reference: reference, chaser: chaser, mailer: mailer)
 
-    references.each do |reference|
-      RefereeMailer.reference_request_chaser_email(reference.application_form, reference).deliver_later
-      ChaserSent.create!(chased: reference, chaser_type: :reminder_reference_nudge)
+      ChaserSent.create!(chased: reference, chaser_type: chaser[:type])
     end
   end
 
-  def send_28_day_chaser!
-    references = ApplicationReference
-      .joins(:application_form)
-      .feedback_requested
-      .where(application_forms: { recruitment_cycle_year: ApplicationForm.select('candidate_id').maximum(:recruitment_cycle_year) })
-      .where(['requested_at < ?', TimeLimitConfig.additional_reference_chase_calendar_days.days.before(Time.zone.now)])
-      .where.not(id: ChaserSent.follow_up_missing_references.select(:chased_id))
+  def deliver_email(reference:, chaser:, mailer:)
+    email_method = mailer.method(chaser[:email])
 
-    references.each do |reference|
-      RefereeMailer.reference_request_chase_again_email(reference).deliver_later
-      CandidateMailer.chase_reference_again(reference).deliver_later
-      ChaserSent.create!(chased: reference, chaser_type: :follow_up_missing_references)
+    return email_method.call(reference, chaser[:extra_params]).deliver_later if chaser[:extra_params].present?
+
+    if chaser[:include_application_form].present?
+      email_method.call(reference.application_form, reference).deliver_later
+    else
+      email_method.call(reference).deliver_later
     end
```

Comment:
suggest
```
    if chaser[:extra_params].present?
      email_method.call(reference, chaser[:extra_params]).deliver_later
    elsif chaser[:include_application_form].present?
      email_method.call(reference.application_form, reference).deliver_later
    else
      email_method.call(reference).deliver_later
    end
```
to reduce the complexity added by the early `return`

Reply:
Done!